src/main/scala/com/cognite/sdk/scala/common/auth.scala

```diff
@@ -45,6 +45,21 @@ final case class BearerTokenAuth(bearerToken: String, override val project: Opti
     r.header("Authorization", s"Bearer $bearerToken")
 }
 
+final case class OIDCTokenAuth(bearerToken: String, override val project: Option[String] = None)
+    extends Auth {
```

Comment:
I could be very well wrong, but why don't we make the `project` required? At least, I think there should be default `None` if it always leads to an exception.

Only problem is that we will probably need to use different identifier than `project` :/

```suggestion
final case class OIDCTokenAuth(bearerToken: String, projectId: String)
    extends Auth {
    override val project = Some(projectId)
```

Reply:
I wanted to keep it in the same style with other Auth types, also thought would be informative if we throw a specific exception in that case.
But if you think the suggestion would be cleaner, I can change it 👍

Reply:
I'm not that sure that my suggestion is that much better. At least I'd remove the default value `None` that will always lead to an error, though.

Reply:
I think it is useful to enforce putting a value there while keeping the same parameters as other Auth types. It will match None case and throw SdkException with the convenient message. 

But I also don't have a strong opinion :D

Reply:
Don't call it `Id`, but `cdfProject` or `projectName` instead. Ids are numbers, except for `externalId`.
Then use it, not `project`, in `def auth` below, to make it impossible to get a runtime error.

Reply:
Thanks! How does it look now?

Reply:
Looks good! :+1: